PhmarcyManagement/app/src/main/java/com/mahibul/phmarcymanagement/ui/DueLIst/view/DueListAdapter.kt

```diff
@@ -0,0 +1,46 @@
+package com.mahibul.phmarcymanagement.ui.DueLIst.view
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import com.mahibul.phmarcymanagement.R
+import com.mahibul.phmarcymanagement.data.reposotory.due_coustomer.DataCustomer
+
+class DueListAdapter(
+        private val customerList : MutableList<DataCustomer>,
+        private val clickListener : CustomerClickListner
+) : RecyclerView.Adapter<DueListViewHolder>() {
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): DueListViewHolder {
+        val view = LayoutInflater.from(parent.context).inflate(R.layout.due_item_view,parent,false)
+        return DueListViewHolder(view)
+    }
+
+    override fun onBindViewHolder(holder: DueListViewHolder, position: Int) {
+        val customer = customerList[position]
+        holder.due_nameTextView.text = customer.customer_name
+        holder.due_priceTextView.text = customer.customer_due.toString()
+
+        holder.due_btnDelete.setOnClickListener {
+            clickListener.onDeleteButtonClicked(customer.customer_name)
+        }
+        holder.due_btnEdit.setOnClickListener {
+            clickListener.onEditButtonClicked(customer.customer_name,customer.customer_due)
+        }
+
+    }
+
+    override fun getItemCount(): Int {
+        return customerList.size
+    }
+
+    fun replaceData(customerList : MutableList<DataCustomer>){
+        this.customerList.clear()
+        this.customerList.addAll(customerList)
+        notifyDataSetChanged()
+    }
+
+    interface CustomerClickListner{
+        fun onEditButtonClicked(customer_name: String,customer_due: Int)
+        fun onDeleteButtonClicked(customer_name: String)
```

Comment:
Do you get the customer using their name? If `yes`, it is a disaster! Use customer ID rather than customer name to uniquely identify them.

Reply:
Yes cause..if identify customer by id name can be conflict... thats why i use customer name as id so that name can be unique...IF there have same name 2 customer he written address with name....

Reply:
Sorry, I don't understand the issue. You can knock me at WhatsApp or call tomorrow morning.